### PR TITLE
test: add MRE for dftatom as integration test case

### PR DESCRIPTION
**Pull request by gxyd**
this helps have the test case directly in *main* instead of *simplifier_pass* branch

### CODE REVIEW

File: integration_tests/binop_of_struct_instance_in_function_call.f90 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+program binop_of_struct_instance_in_function_call
+    use module_binop_of_struct_instance_in_function_call
+    implicit none
+    real(4), target :: A(2), B(2)
+    type(custom_data_type) :: d
+    real(4), dimension(2) :: tmp
+    d%x => A
+    d%y => B
+    tmp = func(d%x - d%y)
```
Comment (Pranavchiku) [inline integration_tests/binop_of_struct_instance_in_function_call.f90:23]: please test value of tmp by passing concrete values of `A` and `B`

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
I think it is fine, you may also test it by assigning values and then doing `if(condition) error stop `

File: integration_tests/binop_of_struct_instance_in_function_call.f90 (hunk 1 above)
Comment (gxyd) [reply integration_tests/binop_of_struct_instance_in_function_call.f90:26]: I've improved the test case and added an `if (...) error stop` condition as well with it.

**Review by certik — APPROVED** [review]

**Comment by gxyd** [comment]
I checked the updated test case as well, that it works with *simplifier_pass* branch.

**Pull request merged**